### DISCUSSION TITLE
[General] <Title> Task oneshot_runner failed when running zksync era node on Linux Dianmu 6.8.0-41-generic

### DISCUSSION THREAD

**Discussion by corerouter** [zkSync-Community-Hub/zksync-developers#699] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

git clone https://github.com/matter-labs/zksync-era.git
cd zksync-era/docs/guides/external-node
cd docker-compose-examples
docker compose --file mainnet-external-node-docker-compose.yml up

mainnet-node-external-node-1     | 2024-09-23T11:30:49.274960Z ERROR zksync_node_framework::service: Task oneshot_runner failed: Oneshot task state_keeper/rocksdb_catchup_task failed

### Repo Link (Optional)

https://github.com/matter-labs/zksync-era

### Additional Details

Running on machine of Linux Dianmu 6.8.0-41-generic #41-Ubuntu SMP PREEMPT_DYNAMIC Fri Aug  2 20:41:06 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

**Comment by alhenaai** [comment]
It appears that you are encountering an error with the `oneshot_runner` task failing during the setup of a ZKsync Era node on Mainnet using Docker. This issue might be related to the node's configuration or a specific problem with the environment setup.

Here are a few steps you can take to troubleshoot and potentially resolve this issue:

1. **Check Docker and Docker Compose Versions:** Ensure that your Docker and Docker Compose are up to date. This can sometimes resolve compatibility issues with the container setup.

2. **Review System Requirements:** Verify that your system meets the hardware requirements necessary for running a ZKsync node. This includes sufficient CPU, RAM, and disk space.

3. **Inspect Logs for More Details:** Look into the Docker container logs for any additional error messages that might give more context on why the `oneshot_runner` task is failing.

4. **Reset Node State:** Sometimes, resetting the node state can help clear any issues that occurred due to previous failed attempts. You can do this by running:
   ```
   cd docker-compose-examples
   docker compose --file mainnet-external-node-docker-compose.yml down --volumes
   ```

5. **Re-run the Node:** After resetting, try running the node again to see if the issue persists.

If these steps do not resolve the issue, I recommend raising this as an issue on the [ZKsync Era GitHub repository](https://github.com/matter-labs/zksync-era) where you can get more specific help from the developers.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by popzxc** [comment, marked as answer]
Hey! `rocksdb_catchup_task` normally fails if your RocksDB cache is out of sync with Postgres. For example, if RocksDB has some data inside, while Postgres is empty. In logs, you should be able to find the exact error message telling what exactly has failed. 

Most likely, RocksDB on your machine (mounted [here](https://github.com/matter-labs/zksync-era/blob/main/docs/guides/external-node/docker-compose-examples/mainnet-external-node-docker-compose.yml#L83)) already has some data. If that's the case, try removing the old data and running the node again.
